shared/wallets/transaction/index.js

```diff
@@ -0,0 +1,178 @@
+// @flow
+import * as React from 'react'
+import {Avatar, Box2, Icon, Text} from '../../common-adapters'
+import {globalColors, globalMargins, platformStyles, styleSheetCreate} from '../../styles'
+// TODO: Format relative dates.
+import {formatTimeForPopup} from '../../util/timestamp'
+
+type Role = 'sender' | 'receiver'
+type CounterpartyType = 'keybaseUser' | 'stellarPublicKey' | 'wallet'
+
+type CounterpartyIconProps = {|
+  large: boolean,
+  counterparty: string,
+  counterpartyType: CounterpartyType,
+|}
+
+const CounterpartyIcon = (props: CounterpartyIconProps) => {
+  const size = props.large ? 48 : 32
+  switch (props.counterpartyType) {
+    case 'keybaseUser':
+      return <Avatar username={props.counterparty} size={size} />
+    case 'stellarPublicKey':
+      // TODO: Use anonymous user icon.
+      return <Icon type="icon-placeholder-avatar-48" style={{width: size, height: size}} />
+    case 'wallet':
+      return <Icon type="icon-wallet-add-48" style={{width: size, height: size}} />
+    default:
+      /*::
+      declare var ifFlowErrorsHereItsCauseYouDidntHandleAllActionTypesAbove: (counterpartyType: empty) => any
+      ifFlowErrorsHereItsCauseYouDidntHandleAllActionTypesAbove(props.counterpartyType);
+      */
+      return null
+  }
+}
+
+type DetailProps = {|
+  large: boolean,
+  yourRole: Role,
+  counterparty: string,
+  counterpartyType: CounterpartyType,
+  amountUser: string,
+|}
+
+const Detail = (props: DetailProps) => {
+  const textType = props.large ? 'Body' : 'BodySmall'
+  const textTypeSemibold = props.large ? 'BodySemibold' : 'BodySmallSemibold'
+
+  let counterparty
+  switch (props.counterpartyType) {
+    case 'keybaseUser':
+      // TODO: Color counterparty based on following status.
+      counterparty = <Text type={textTypeSemibold}>{props.counterparty}</Text>
```

Comment:
this exists, just use common-adaters/usernames

Reply:
Oh yeah, done